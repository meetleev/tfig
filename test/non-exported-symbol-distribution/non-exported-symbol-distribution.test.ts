
import ps from 'path';
import fs from 'fs-extra';
import { bundle, IOptions } from '../../source/gift';

test('Non exported symbol distribution', async () => {
    const inputPath = ps.join(__dirname, 'input.d.ts');
    const outputPath = ps.join(__dirname, 'output.d.ts');

    const bundleOptions: IOptions = {
        input: inputPath,
        output: outputPath,
        entries: {
            'baz': 'baz', // The firstly encountered module; we processing module in order of entries.
            'foo': 'foo',
            'bar': 'bar',
        },
    };

    {
        const { groups } = bundle({
            ...bundleOptions,
            nonExportedSymbolDistribution: [{
                sourceModule: /internal/,
                targetModule: 'foo',
            }],
        });
        expect(groups.length).toBe(1);
        const group0 = groups[0];
        expect(group0.code).toMatchSnapshot('Export to foo');
    }

    {
        const { groups } = bundle({
            ...bundleOptions,
            nonExportedSymbolDistribution: [{
                sourceModule: /internal/,
                targetModule: 'bar',
            }],
        });
        expect(groups.length).toBe(1);
        const group0 = groups[0];
    
        expect(group0.code).toMatchSnapshot('Export to bar');
    }

    {
        // By default, export to "current" module(where it's firstly encountered)
        const { groups } = bundle({
            ...bundleOptions,
        });
        expect(groups.length).toBe(1);
        const group0 = groups[0];
    
        expect(group0.code).toMatchSnapshot('Export to "firstly encountered" module');
    }
});