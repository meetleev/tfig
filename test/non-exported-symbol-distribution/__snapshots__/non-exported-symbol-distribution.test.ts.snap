// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Non exported symbol distribution: Export to "firstly encountered" module 1`] = `
"declare module \\"baz\\" {
    export function foo(): __private._internal__NonExported;
    export namespace __private {
        export class _internal__NonExported {
        }
    }
    export {};
}
declare module \\"foo\\" {
    export function foo(): ___private._internal__NonExported;
    import { __private as ___private } from \\"baz\\";
    export {};
}
declare module \\"bar\\" {
    export function bar(): ___private._internal__NonExported;
    import { __private as ___private } from \\"baz\\";
    export {};
}
"
`;

exports[`Non exported symbol distribution: Export to bar 1`] = `
"declare module \\"baz\\" {
    export function foo(): ___private._internal__NonExported;
    import { __private as ___private } from \\"bar\\";
    export {};
}
declare module \\"foo\\" {
    export function foo(): ___private._internal__NonExported;
    import { __private as ___private } from \\"bar\\";
    export {};
}
declare module \\"bar\\" {
    export function bar(): __private._internal__NonExported;
    export namespace __private {
        export class _internal__NonExported {
        }
    }
    export {};
}
"
`;

exports[`Non exported symbol distribution: Export to foo 1`] = `
"declare module \\"baz\\" {
    export function foo(): ___private._internal__NonExported;
    import { __private as ___private } from \\"foo\\";
    export {};
}
declare module \\"foo\\" {
    export function foo(): __private._internal__NonExported;
    export namespace __private {
        export class _internal__NonExported {
        }
    }
    export {};
}
declare module \\"bar\\" {
    export function bar(): ___private._internal__NonExported;
    import { __private as ___private } from \\"foo\\";
    export {};
}
"
`;
